src/modules/auth/controllers/het-archief.controller.ts

```diff
@@ -226,6 +228,22 @@ export class HetArchiefController {
 
 			SessionHelper.setArchiefUserInfo(session, archiefUser);
 
+			// Update custom fields in Campaign Monitor
+			try {
+				await this.campaignMonitorService.updateNewsletterPreferences({
+					firstName: archiefUser?.firstName,
+					lastName: archiefUser?.lastName,
+					email: archiefUser?.email,
+					is_key_user: archiefUser?.isKeyUser,
+					usergroup: archiefUser?.groupName,
+					created_date: archiefUser?.createdAt,
+					last_access_date: archiefUser?.lastAccessAt,
+					organisation: archiefUser?.organisationName,
+				});
+			} catch (err) {
+				this.logger.error('Failed updating the custom fields to Campaign Monitor');
+			}
```

Comment:
This is an example where we don't want to wait for campaign monitor to allow the user to login.
Its already good that you added a try catch. But if something goes wrong with campaign monitor and requests have to timeout before they fail, the login could take a long time.
So we better not wait, but still catch the error. You can do that like this:
```suggestion
			
				this.campaignMonitorService.updateNewsletterPreferences({
					firstName: archiefUser?.firstName,
					lastName: archiefUser?.lastName,
					email: archiefUser?.email,
					is_key_user: archiefUser?.isKeyUser,
					usergroup: archiefUser?.groupName,
					created_date: archiefUser?.createdAt,
					last_access_date: archiefUser?.lastAccessAt,
					organisation: archiefUser?.organisationName,
				}).catch(err => {
					this.logger.error('Failed updating the custom fields to Campaign Monitor: ' + JSON.stringify(err));
				})
```